robot_name: 'go2'

joint_names: &jn ["LF_HAA", "LF_HFE", "LF_KFE", "LH_HAA", "LH_HFE", "LH_KFE", "RF_HAA", "RF_HFE", "RF_KFE", "RH_HAA", "RH_HFE", "RH_KFE"]
base_name: "base"

state_topic: "go2/state"
command_topic: "go2/command"
change_controller_topic: "go2/change_controller"
contact_topics: ["/lf_foot_contact", "/rf_foot_contact", "/lh_foot_contact","/rh_foot_contact"] 

use_lidar: true

marker_topic: "markers"
odom_frame: "odom"

static_controller:
  stand_controller:
    joint_angles: [0.0, 0.806, -1.802, 0.0, 0.996, -1.802, 0.0, 0.806, -1.802, 0.0, 0.996, -1.802] # LF, LH, RF, RH
  sit_controller:
    joint_angles: [0.0, 1.36, -2.55, 0.0, 1.36, -2.55, 0.0, 1.36, -2.55, 0.0, 1.36, -2.55] # LF, LH, RF, RH
  kp: 90
  kd: 0.5
  interpolation_time: 1.0 # s
  rate: 50 # Hz
  publish_state: true

np3o:
  kp: 40.0
  kd: 1.0
  use_action_filter: true

  publish_state: true

  hf_repo: "kubjonai/tbai-models"
  hf_model: "np3o_model3_put_feet_higher.pt" #  np3o_model_1_original.pt, np3o_model_2_higher_action_rate_cost.pt, np3o_model3_put_feet_higher.pt

wtw_controller:
  kp: 40.0
  kd: 1.0
  publish_state: true
  state_topic: "estimated_state"

  hf_repo: "kubjonai/tbai-models"
  hf_adaptation_module: "wtw_adaptation_module_latest.jit"
  hf_body_module: "wtw_body_latest.jit"

bob_controller:
  blind: true
  kp: 30.0
  kd: 0.5

  # https://huggingface.co/kubjonai/tbai-models/tree/main
  # For go2_blind.pt, set lin_vel_scale to 2.0, 0.0 for other models
  hf_repo: "kubjonai/tbai-models"
  hf_model: "go2_blind.pt" # go2_blind.pt,  go2_blind_no_vel.pt,  go2_blind_no_vel_noise.pt

  action_scale: 0.25
  lin_vel_scale: 2.0 # 0.0 for no velocity
  publish_state: true

  default_stance: [
    [ 0.025,  0.096, -0.263],
    [-0.025,  0.096, -0.263],
    [ 0.025, -0.096, -0.263],
    [-0.025, -0.096, -0.263],
  ]

  ik:
    d2: 0.0955
    a3: 0.213
    a4: 0.213

  cpg:
    period: 0.6
    time_offsets: [0.0, 0.3, 0.3, 0.0]  # LF, LH, RF, RH
    swing_height: 0.1

reference_generator:
  type: "twist" # twist, joystick, go2_joystick

  joystick:
    topic: "joy"
    ramped_velocity: 5.0
    x_index: 1
    y_index: 0
    yaw_index: 3
    x_scale: 1.0
    y_scale: 0.5
    yaw_scale: 0.5

  twist:
    topic: "cmd_vel"

inekf_estimator:
  rectify_orientation: true
  remove_gyroscope_bias: true

go2_joystick:
  velocity_factor_x: 0.6
  velocity_factor_y: 0.5
  yaw_rate_factor: 0.4
